src/com/zebrunner/jenkins/pipeline/tools/scm/github/GitHub.groovy

```diff
@@ -1,153 +1,53 @@
 package com.zebrunner.jenkins.pipeline.tools.scm.github
 
-import com.zebrunner.jenkins.Logger
-import com.zebrunner.jenkins.pipeline.tools.scm.ISCM
+import com.zebrunner.jenkins.pipeline.tools.scm.Scm
 import com.zebrunner.jenkins.pipeline.Configuration
-import static com.zebrunner.jenkins.pipeline.Executor.*
-import static com.zebrunner.jenkins.Utils.*
 
-class GitHub implements ISCM {
+class GitHub extends Scm {
 
-    protected def context
-    //TODO: rename into scmUrl as it covers both https and ssh cases
-    protected def gitHtmlUrl
-    protected def credentialsId
-    protected Logger logger
-    protected def scmHost
-
-    public GitHub(context) {
-        this.context = context
-        logger = new Logger(context)
-        scmHost = Configuration.get(Configuration.Parameter.GITHUB_HOST)
-        if (scmHost.contains("upnetix")) {
-            gitHtmlUrl = "https://\${GITHUB_HOST}/scm/\${GITHUB_ORGANIZATION}/${Configuration.get("repo")}"
-        } else if (scmHost.contains("bitbucket")) {
-            gitHtmlUrl = "https://\${GITHUB_HOST}/\${GITHUB_ORGANIZATION}/${Configuration.get("repo")}"
-        } else if (scmHost.contains("gitlab")) {
-            gitHtmlUrl = "https://\${GITHUB_HOST}/\${GITHUB_ORGANIZATION}/${Configuration.get("repo")}"
-        } else if (scmHost.contains("io")) {
```

Comment:
one of the example when we can't determine scm type by hostname:(

Reply:
What does io means? (custom scm server ?) if it behaves like github/gitlab/bitbucket(in the api point of view(to check for the Json response to configure the webhooks jobs))